File: nomad-samples/js/asset-upload/assets/start-asset-upload.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function startUpload(authToken, PARENT_ID, CONTENT_LENGTH, UPLOAD_OVERWRITE_OPTION, CHUNK_SIZE, RELATIVE_PATH, LANGUAGE_ID) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${authToken}`);
  
    const BODY = {
        "parentId": PARENT_ASSET_ID,
        "contentLength": CONTENT_LENGTH,
        "uploadOverwriteOption": UPLOAD_OVERWRITE_OPTION,
        "chunkSize": CHUNK_SIZE,
        "relativePath": RELATIVE_PATH,
        "languageId": LANGUAGE_ID
    };

    // Post
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/asset/upload/start`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    apiExceptionHandler(RESPONSE, "Start asset upload failed");
}

File: nomad-samples/python/Asset Uploads/assets/start_asset_upload.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def start_upload(AUTH_TOKEN: str) -> dict:

    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")

    API_URL = ADMIN_API_URL + "/asset/upload/start"
        
    # Create header for the request
    HEADERS = {
  	    "Authorization": "Bearer " + AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    # Build the payload BODY
    BODY = {
      	"parentId":"73d06e60-9607-4018-b666-775790c0f0c2",
      	"contentLength":3136083,
      	"uploadOverwriteOption":"cancel",
      	"chunkSize":8388608,
      	"relativePath":"test.txt",
        "languageId":"c66131cd-27fc-4f83-9b89-b57575ac0ed8"
    }

    try:
        # Send the request
        RESPONSE = requests.post(API_URL, headers= HEADERS, data= json.dumps(BODY))

        if RESPONSE.status_code != 200:
            raise Exception("Response returned " + str(RESPONSE.status_code))

        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "Start asset upload failed")

